host/4/bookworm/python/python312/python312-appservice.Dockerfile

```diff
@@ -0,0 +1,112 @@
+    rm -f /$EXTENSION_BUNDLE_FILENAME_V4 &&\
+    find /FuncExtensionBundles/ -type f -exec chmod 644 {} \;
+
+FROM mcr.microsoft.com/oryx/python:3.12-debian-bookworm AS python
+
+# Install Python dependencies
+# MS SQL related packages: unixodbc msodbcsql18 mssql-tools
+# .NET Core dependencies: --no-install-recommends ca-certificates libc6 libgcc1 libgssapi-krb5-2 libicu67 libssl1.1 libstdc++6 zlib1g
+# OpenCV dependencies:libglib2.0-0 libsm6 libxext6 libxrender-dev xvfb
+# binutils: binutils
+# OpenMP dependencies: libgomp1 && \
+# Fix from https://github.com/GoogleCloudPlatform/google-cloud-dotnet-powerpack/issues/22#issuecomment-729895157 : libc-dev
+# Azure ML dependencies: liblttng-ust0
+RUN curl -fsSL https://packages.microsoft.com/keys/microsoft.asc | gpg --dearmor -o /usr/share/keyrings/microsoft-prod.gpg && \
+    apt-get update && \
+    apt-get install -y wget apt-transport-https curl gnupg2 locales rpm && \
+    echo 'debconf debconf/frontend select Noninteractive' | debconf-set-selections && \
+    curl https://packages.microsoft.com/keys/microsoft.asc | tee /etc/apt/trusted.gpg.d/microsoft.asc && \ 
+    echo "deb [arch=amd64] https://packages.microsoft.com/debian/12/prod bookworm main" | tee /etc/apt/sources.list.d/mssql-release.list && \
+    # Needed for libss1.0.0 and in turn MS SQL
+    echo 'deb http://security.debian.org/debian-security bookworm-security main' >> /etc/apt/sources.list && \
+    curl https://packages.microsoft.com/config/debian/12/prod.list | tee /etc/apt/sources.list.d/mssql-release.list && \
+    # install MS SQL related packages.pinned version in PR # 1012.
+    echo 'en_US.UTF-8 UTF-8' > /etc/locale.gen && \
+    locale-gen && \
+    apt-get update && \
+    ACCEPT_EULA=Y apt-get install -y unixodbc msodbcsql18 mssql-tools18 &&\
+    apt-get install -y --no-install-recommends ca-certificates \
+    libc6 libgcc1 libgssapi-krb5-2 libicu72 libssl3 libstdc++6 zlib1g &&\
+    apt-get install -y libglib2.0-0 libsm6 libxext6 libxrender-dev xvfb binutils \
+    binutils libgomp1 libc-dev liblttng-ust1 && \
+    rm -rf /var/lib/apt/lists/* 
+
+FROM mcr.microsoft.com/dotnet/aspnet:8.0-bookworm-slim-amd64
+ARG HOST_VERSION
+
+COPY --from=runtime-image [ "/usr/share/dotnet", "/usr/share/dotnet" ]
+COPY --from=runtime-image ["/azure-functions-host", "/azure-functions-host"]
+COPY --from=runtime-image [ "/FuncExtensionBundles", "/FuncExtensionBundles" ]
+COPY start.sh /azure-functions-host/
+COPY install_ca_certificates.sh start_nonappservice.sh /opt/startup/
+RUN chmod +x /opt/startup/install_ca_certificates.sh && \
+    chmod +x /opt/startup/start_nonappservice.sh && \
+    chmod +x /azure-functions-host/start.sh
+
+COPY --from=runtime-image [ "/workers/python/3.12/LINUX", "/azure-functions-host/workers/python/3.12/LINUX" ]
+COPY --from=runtime-image [ "/workers/python/worker.config.json", "/azure-functions-host/workers/python" ]
+COPY --from=python [ "/", "/" ]
```

Comment:
The only larger piece of feedback I have is that I think we should be more specific about what we install to the runtime image. If we copy over the entirety of the image we hide a lot of copy actions behind one statement. Also, we overwrite anything pre-existing in the image with the context from the python image. This means that if we installed some key or library in the host image but a different one is in the oryx image we may overwrite that key or content without knowing. I think we can be more specific by following this approach  https://msazure.visualstudio.com/One/_git/AAPT-Antares-Functions-Docker?path=/images/flexconsumption/4-dotnet8/runtimes/python310.Dockerfile&version=GBdev&line=31&lineEnd=32&lineStartColumn=2&lineEndColumn=1&lineStyle=plain&_a=contents

As you can see in this image definition, we install the python dependencies into the final runtime image to simplify the python copy. By doing this, we move to getting ONLY python from the base image

Reply:
Fixed by refactoring FROM commands